### ISSUE TITLE
The application being debugged crashes with the OpenGL context.

### ISSUE THREAD

**Issue by LaG1924** [baldurk/renderdoc#2302]
## Description

Installed the current stable version of this debugger. There were no problems with my application a couple of years ago. Now my application is crashing, but one frame with the application interface is being rendered.
I use SDL to manage windows. Creating an OpenGL context in a window causes the process to close after two frames (no error messages from the OS). The usual launch of a separate process or from the VS debugger does not lead to problems (an empty black window appears and remains).
VS debugger shows memory error inside Nvidia driver (inside dll file starting with 'nv').
Also I tried toggling all capture options but that also didn't help.
Minimum reproducible piece of code:
```
int main(int argc, char** argv) {
    if (SDL_Init(SDL_INIT_VIDEO) < 0)
        exit(-1);

    SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 3);
    SDL_Window* window = SDL_CreateWindow(
        "MyApp", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
        900, 480, SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE);
    if (!window)
        exit(-1);

    SDL_GLContext glContext = SDL_GL_CreateContext(window);
    if (!glContext)
        exit(-1);

    while (true) {
        SDL_PumpEvents();
        SDL_Event event;
        while (SDL_PollEvent(&event)) { if (event.type == SDL_WINDOWEVENT && event.window.event == SDL_WINDOWEVENT_CLOSE) exit(-2); }
        SDL_GL_SwapWindow(window);
    }	
	return 0;
}
```
## Steps to reproduce
1. Download any artifact [here](https://github.com/LaG1924/AltCraft/actions/runs/975552850) or the [Crash3.zip](https://github.com/baldurk/renderdoc/files/6721220/Crash3.zip) (it contains the same project in which I left only the above code).
2. Run without RenderDoc to make sure it doesn't crash.
3. Run with RenderDoc and see that it will crash immediately after it launches.

## Environment

* RenderDoc version: v1.14
* Operating System: Win10-21H1
* Graphics API: OpenGL 3.3
* GPU: RTX 3080
* GPU Driver version: 466.11

**Comment by baldurk** [comment]
I can't reproduce this crash. I didn't compile the code, but I ran both the artifact and Crash3.zip you linked and both run correctly for me both inside and outside of RenderDoc with no crashes and I'm able to take captures of both.

Assuming you don't get a crash dialog popup from RenderDoc to submit a crash, it sounded like you were able to attach visual studio to the crash. Can you share the callstack and a minidump manually? Can you also share the RenderDoc log please?

**Comment by LaG1924** [comment]
`Exception thrown at 0x000001DCCB81FDC0 in AltCraft.exe: 0xC0000005: Access violation reading location 0x0000000000000000.`
Exception callstack (this is useless I think):
```
000001dccb81fdc0()
00000000044323a0()
```
Thread with exception: `[2380] nvoglv64.dll thread`

Main thread in window framebuffers switching function:
```
>	SDL2d.dll!WIN_GL_SwapWindow(SDL_VideoDevice * _this, SDL_Window * window) Line 855	C
 	SDL2d.dll!SDL_GL_SwapWindow_REAL(SDL_Window * window) Line 3741	C
 	SDL2d.dll!SDL_GL_SwapWindow(SDL_Window * a) Line 594	C
 	AltCraft.exe!SDL_main(int argc, char * * argv) Line 57	C++
 	AltCraft.exe!main_getcmdline() Line 71	C
 	AltCraft.exe!main(int argc, char * * argv) Line 87	C
 	[External Code]	
```

Unfortunately, I cannot upload minidump with heap (due to size I think: 63MB)
[Without heap.zip](https://github.com/baldurk/renderdoc/files/6721692/Without.heap.zip)
[renderdoc_log.log](https://github.com/baldurk/renderdoc/files/6721700/renderdoc_log.log)

**Comment by baldurk** [comment]
Since the crash happens inside nvidia's driver thread and I can't reproduce it, this doesn't seem like a bug in RenderDoc. You have some programs installed such as RivaTuner Statistics Server and something unknown called A-Volute. At least the RivaTuner program I have seen be _extremely_ buggy in the past so I'd recommend trying to uninstall or disable them both to see if that fixes the problem.

Otherwise I'd suggest seeing if there's an updated driver or doing a clean reinstall with DDU. If that still doesn't work try on a different machine if you are able to, in case it's some systemic issue.

**Comment by LaG1924** [comment]
Thanks for the help anyway.

I was expecting a similar answer, so I kept looking for a solution... and found it.
I tried: updating drivers (stable and beta), disabling (and then completely removing) RivaTuner (maybe there were bugs in the past, but I have been using RTSS all the time for the last few years and only a couple of old games had problems), updating all OS components.
And then you wrote about A-Volute. I found that this is part of the sound driver from my motherboard (asus rog strix b450-e gaming) and there are two utilities in the system from A-Volute (to improve the sound if I understand correctly) Sonic Radar 3 and Sonic Studio 3. After removing them, the problem went away!
